src/multipart_.rs

```diff
@@ -314,7 +336,12 @@ fn header(name: &str, field: &Part) -> String {
         match field.mime {
             Some(ref mime) => format!("\r\nContent-Type: {}", mime),
             None => "".to_string(),
-        }
+        },
+        field.hdr.iter().fold(
+            "".to_string(),
+            |hdrs, (k,v)|
+                format!("{}\r\n{}: {}", hdrs, k.as_str(), v)
```

Comment:
This will actually allocate a new string for each header, whereas they could just be pushed onto the existing one... Though, there shouldn't be that many for a single part, so it's probably not a big deal.

Reply:
I hope + on String does what I think it does.